### PR TITLE
implement mint function

### PR DESCRIPTION
**Pull request by octate**
@idir, I have added new param in mint function definition `_uri` and have changed the validation from !exists to exists for tokenId. If they does not make sense, let me know

### CODE REVIEW

File: evm_bridged_nft/contracts/BridgedNFT.sol [hunk 1]
```diff
@@ -28,11 +28,16 @@ contract BridgedNFT is ERC721Upgradeable {
...
     /// new nft token
-    /// @dev check if the token id not exists, then mint a new one by calling _mint function
+    /// @dev check if the token id exists, then mint a new one by calling _mint function
     /// inherited from the ERC721Upgradeable then pass _recipient and _tokenId.
     /// @param _tokenId nft token id.
     /// @param _recipient owner of the nft.
-    function mintNFT(uint256 _tokenId, address _recipient) external {}
+    /// @param _uri setting token URI using token id and URI of JSON file.
+    function mintNFT(uint256 _tokenId, address _recipient, string _uri) external {
+        require(_exists(_tokenId));
```
Comment (idirall22) [inline evm_bridged_nft/contracts/BridgedNFT.sol]: You are checking if the token exists if yes to should fail, so you have to add "!" before _exists function.
here add as 2nd args a string to tell why it fails.

File: evm_bridged_nft/contracts/BridgedNFT.sol [hunk 2]
```diff
@@ -28,11 +28,16 @@ contract BridgedNFT is ERC721Upgradeable {
...
+        super._mint(_recipient, _tokenId);
+        super._setTokenURI(_tokenId, _uri);
```
Comment (idirall22) [inline evm_bridged_nft/contracts/BridgedNFT.sol:40]: Where I can found this  "_setTokenURI" function??

**Review by idirall22 — CHANGES_REQUESTED** [review]

File: evm_bridged_nft/contracts/BridgedNFT.sol (hunk 2 above)
Comment (idirall22) [reply evm_bridged_nft/contracts/BridgedNFT.sol:40]: Also please add tests

Comment (octate) [reply evm_bridged_nft/contracts/BridgedNFT.sol:40]: https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721Metadata-_setTokenURI-uint256-string-

**Pull request closed without merging**